python/slope_api.py

```diff
@@ -0,0 +1,118 @@
+        response = requests.put(uploadUrl, data=open(filename, "rb"))
+        self.check_response(response)
+
+        response = self.session.post(f"{self.api_url}/Files/SaveUpload", json=slope_file_params)
+        self.check_response(response)
+        return response.json()["fileId"]
+        
+    def create_data_table(self, filename: str, slope_table_params) -> int:
+        self.upload_file(filename, slope_table_params["filePath"])
+        response = self.session.post(f"{self.api_url}/DataTables", json=slope_table_params)
+        self.check_response(response)
+        return response.json()["id"]
+    
+    def create_scenario_table(self, filename: str, slope_scenario_table_params) -> int:
+        self.upload_file(filename, slope_scenario_table_params["filePath"])
+        response = self.session.post(f"{self.api_url}/ScenarioTables", json=slope_scenario_table_params)
+        self.check_response(response)
+        return response.json()["id"]
+    
+    def create_projection_from_template(self, template_id: int, name: str) -> int:
+        params = {"templateId": template_id, "name": name}
+        response = self.session.post(f"{self.api_url}/Projections", json=params)
+        self.check_response(response)
+        return response.json()["id"]
+    
+    def update_projection(self, projection_id, properties):
+        response = self.session.patch(f"{self.api_url}/Projections/{projection_id}", json=properties)
+        self.check_response(response)
+        
+    def update_projection_mpf(self, projection_id, portfolio_name, product_name, model_point_file_id):
+        projection_update_parameters = {
+            "portfolios": [{
+                "portfolioName": portfolio_name,
+                "products": [{
+                    "productName": product_name,
+                    "modelPointFile": {
+                        "fileId": model_point_file_id
+                    }
+                }]
+            }]        
+        }
+        response = self.session.patch(f"{self.api_url}/Projections/{projection_id}", json=projection_update_parameters)
+        self.check_response(response)
+        
+    def update_projection_table(self, projection_id, table_name, data_table_id):
+        projection_update_parameters = {
+            "dataTables": [{
+                "tableStructureName": table_name,
+                "dataTableId": data_table_id
+            }]      
+        }
+        response = self.session.patch(f"{self.api_url}/Projections/{projection_id}", json=projection_update_parameters)
+        self.check_response(response)
+    
+    def run_projection(self, projection_id):
+        response = self.session.post(f"{self.api_url}/Projections/{projection_id}/run")
+        if not response.ok:
+            raise Exception(f"Failed to start projection with id: {projection_id}", response.text)
+
+    def is_projection_running(self, projection_id) -> bool:
```

Comment:
One note:  We actually pass back an IsRunning flag from our API now, and I think we should encourage API consumers to use that instead of checking the statuses manually.  Can we update this to use that field?